google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/CloudSdkAppEngineHelper.java

```diff
@@ -94,26 +98,27 @@ public Project getProject() {
 
   @NotNull
   @Override
-  public Path defaultAppYaml() {
-    return getFileFromResourcePath(DEFAULT_APP_YAML_PATH);
+  public Optional<Path> defaultAppYaml() {
+    return Optional.of(getFileFromResourcePath(DEFAULT_APP_YAML_PATH));
   }
 
   @Nullable
   @Override
-  public Path defaultDockerfile(AppEngineFlexDeploymentArtifactType deploymentArtifactType) {
+  public Optional<Path> defaultDockerfile(
+      AppEngineFlexibleDeploymentArtifactType deploymentArtifactType) {
     switch (deploymentArtifactType) {
       case WAR:
-        return getFileFromResourcePath(DEFAULT_WAR_DOCKERFILE_PATH);
+        return Optional.of(getFileFromResourcePath(DEFAULT_WAR_DOCKERFILE_PATH));
       case JAR:
-        return getFileFromResourcePath(DEFAULT_JAR_DOCKERFILE_PATH);
+        return Optional.of(getFileFromResourcePath(DEFAULT_JAR_DOCKERFILE_PATH));
       default:
-        return null;
+        return Optional.empty();
     }
   }
 
   @Nullable
```

Comment:
Good catch. Actually, with Optionals, we can get rid of @Nullable/@NotNull annotations altogether (where Optionals are used), since they'll always be @NotNull.